src/cli/telemetry/telemetry.ts

```diff
@@ -0,0 +1,309 @@
+  const targets = await getActiveTargets(validatedConfig.config);
+  const versions = getCoreCompiler()?.versions || { typescript: 'unknown', rollup: 'unknown' };
+  const component_count = Object.keys(result.componentGraph).length;
+
+  const data: TrackableData = {
+    session_id,
+    yarn: isUsingYarn(),
+    duration_ms: result.duration,
+    component_count,
+    targets,
+    packages: packages || [],
+    arguments: flags.args,
+    task: flags.task,
+    stencil: getCoreCompiler()?.version || 'unknown',
+    system: `${sys.name} ${sys.version}`,
+    os_name: details.platform,
+    os_version: details.release,
+    cpu_model: `${details.cpuModel}`,
+    build: getCoreCompiler()?.buildId || 'unknown',
+    typescript: versions.typescript,
+    rollup: versions.rollup,
+  };
+
+  await sendMetric('stencil_cli_command', data);
+  logger.debug(`${logger.blue('Telemetry')}: ${logger.gray(JSON.stringify(data))}`);
+}
+
+/**
+ * A function to wrap a compiler task function around. Will send telemetry if, and only if, the machine allows.
+ * @param action A Promise-based function to call in order to get the duration of any given command.
+ * @returns void
+ */
+export async function telemetryAction(action?: TelemetryCallback) {
+  const { sys, flags, logger, validatedConfig } = getStencilCLIConfig();
+  const tracking = await shouldTrack(!!flags.ci);
+
+  const details = sys.details;
+  const versions = getCoreCompiler()?.versions || { typescript: 'unknown', rollup: 'unknown' };
+  let duration = undefined;
+  let error: any;
+
+  if (action) {
+    const start = new Date();
+
+    try {
+      await action();
+    } catch (e) {
+      error = e;
+    }
+
+    const end = new Date();
+    duration = end.getTime() - start.getTime();
+  }
+
+  // We'll get componentCount details inside the taskBuild, so let's not send two messages.
+  if (!tracking || (flags.task == 'build' && !flags.args.includes('--watch'))) {
+    return;
+  }
+
+  const session_id = await getTelemetryToken();
+  const packages = await getInstalledPackages();
+  const targets = await getActiveTargets(validatedConfig.config);
+
+  const data: TrackableData = {
+    session_id,
+    yarn: isUsingYarn(),
+    duration_ms: duration,
+    targets,
+    packages: packages || [],
+    arguments: flags.args,
+    task: flags.task,
+    stencil: getCoreCompiler()?.version || 'unknown',
+    system: `${sys.name} ${sys.version}`,
+    os_name: details.platform,
+    os_version: details.release,
+    cpu_model: `${details.cpuModel}`,
+    build: getCoreCompiler()?.buildId || 'unknown',
+    typescript: versions.typescript,
+    rollup: versions.rollup,
+  };
+
+  await sendMetric('stencil_cli_command', data);
+  logger.debug(`${logger.blue('Telemetry')}: ${logger.gray(JSON.stringify(data))}`);
+
+  if (error) {
+    throw error;
+  }
+}
+
+function isUsingYarn() {
+  return getCompilerSystem().getEnvironmentVar('npm_execpath').includes('yarn');
```

Comment:
This is throwing `TypeError: Cannot read property 'includes' of undefined` when the CLI is invoked directly.
```suggestion
  return getCompilerSystem().getEnvironmentVar('npm_execpath')?.includes('yarn') || false;
```